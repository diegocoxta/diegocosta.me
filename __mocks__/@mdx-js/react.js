module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
