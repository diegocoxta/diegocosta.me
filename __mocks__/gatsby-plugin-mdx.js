module.exports = {
  MDXRenderer: jest.fn(({ children }) => children),
};
