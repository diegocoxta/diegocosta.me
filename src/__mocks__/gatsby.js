const React = require('react');

module.exports = {
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({ to, ...rest }) =>
      React.createElement('a', {
        ...rest,
        href: to,
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
  navigate: jest.fn(),
};
