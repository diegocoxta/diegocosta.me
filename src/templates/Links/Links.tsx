import React from 'react';
import styled from 'styled-components';

import Footer from '~/components/Footer';
import BrandName from '~/components/BrandName';
import ThemeProvider, { GlobalStyle } from '~/components/ThemeProvider';

const Container = styled.section`
  max-width: 660px;
  margin: 0 auto;
  padding: 0 20px;
`;

const Avatar = styled.img`
  width: 90px;
  height: 90px;
`;

const GroupTitle = styled.h2`
  text-transform: uppercase;
  text-align: center;
  font-size: 16px;
  font-weight: 700;
  margin-bottom: 20px;
`;

const ExternalLink = styled.a`
  border: 1px solid #d4d4d8;
  padding: 5px;
  border-radius: 100px;
  display: block;

  display: flex;
  justify-content: flex-start;

  margin-bottom: 20px;
`;

const Bullet = styled.div`
  width: 50px;
  height: 50px;
  border-radius: 50px;
  background-color: ${(props) => props.theme.accentColor};
`;

const Label = styled.p`
  margin: auto 10px;
`;

export interface LinksProps {
  author: string;
  avatar: string;
  description: string;
}

export default function Links(props: LinksProps): React.ReactElement {
  return (
    <ThemeProvider>
      <GlobalStyle />
      <Container>
        <Avatar src={props.avatar} />
        <BrandName author={props.author} size="small" href="/" />
      </Container>
      <Container>
        <GroupTitle>Featured Links</GroupTitle>
        {[0, 0, 0, 0, 0].map((i, index) => (
          <ExternalLink key={index}>
            <Bullet />
            <Label>My Website</Label>
          </ExternalLink>
        ))}

        <GroupTitle>Other Links</GroupTitle>
        {[0, 0, 0, 0, 0].map((i, index) => (
          <ExternalLink key={index}>
            <Bullet />
            <Label>My Website</Label>
          </ExternalLink>
        ))}

        <p>Content</p>
        <Footer />
      </Container>
    </ThemeProvider>
  );
}
