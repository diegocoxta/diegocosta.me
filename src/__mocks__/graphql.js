import * as Gatsby from 'gatsby';

export const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);
export const mockUseStaticQuery = {
  site: {
    siteMetadata: {
      name: 'Diego Costa',
      website: {
        title: 'My Website',
      },
    },
  },
  pages: {
    nodes: [{}],
  },
};

export const mockPageQuery = {
  list: {
    totalCount: 2,
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Title 1',
            date: '2021-01-16T22:12:03.284Z',
            tags: [],
            description: 'Awesome first article',
          },
          fields: {
            slug: '/path-to-article',
            readingTime: {
              minutes: 5,
            },
          },
          excerpt: 'Awesome first article',
          html: 'Post content',
        },
      },
      {
        node: {
          frontmatter: {
            title: 'Title 2',
            date: '2020-01-18T22:12:03.284Z',
            tags: [],
            description: 'Awesome second article',
          },
          fields: {
            slug: '/path-to-article-2',
            readingTime: {
              minutes: 2,
            },
          },
          excerpt: 'Awesome first article',
          html: 'Post content #2',
        },
      },
    ],
  },
  articlesMdx: {
    edges: [],
  },
};
