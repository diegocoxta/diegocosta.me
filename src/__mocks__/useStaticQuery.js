import * as Gatsby from 'gatsby';

export const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);
export const mockUseStaticQuery = {
  site: {
    siteMetadata: {
      metatags: {
        title: `Gatsby Default Starter`,
        author: 'Diego Costa',
      },
    },
  },
  pages: {
    nodes: [{}],
  },
};
