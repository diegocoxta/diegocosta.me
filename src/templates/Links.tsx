import React from 'react';
import styled from 'styled-components';

import Footer from '~/components/Footer';
import Logo from '~/components/Logo';
import { ThemeProvider, GlobalStyle } from '~/components/ThemeSwitcher';
import { graphql, useStaticQuery } from 'gatsby';
import Metatags from '~/components/Metatags';
import Divisor from '~/components/Divisor';

const Container = styled.section`
  max-width: 660px;
  margin: 0 auto;
  padding: 0 20px;
`;

const Avatar = styled.img`
  width: 90px;
  height: 90px;
`;

const GroupTitle = styled.h2`
  text-transform: uppercase;
  text-align: center;
  font-size: 16px;
  font-weight: 700;
  margin-bottom: 20px;
`;

const ExternalLink = styled.a`
  border: 1px solid #d4d4d8;
  padding: 5px;
  border-radius: 100px;
  display: block;

  display: flex;
  justify-content: flex-start;

  margin-bottom: 20px;
`;

const Bullet = styled.div`
  width: 50px;
  height: 50px;
  border-radius: 50px;
  background-color: ${(props) => props.theme.accentColor};
`;

const Label = styled.p`
  margin: auto 10px;
`;

export default function Links(): React.ReactElement {
  const {
    site: {
      siteMetadata: {
        sourceCode,
        metatags: { title, author, avatar, description, banner },
      },
    },
  } = useStaticQuery(graphql`
    query LinksTemplateQuery {
      site {
        siteMetadata {
          sourceCode
          metatags {
            title
            author
            avatar
            description
            banner
          }
        }
      }
    }
  `);

  return (
    <ThemeProvider>
      <GlobalStyle />
      <Metatags author={author} banner={banner} title={title} description={description} />
      <Container>
        <Avatar src={avatar} />
        <Logo author={author} size="small" />
      </Container>
      <Divisor />
      <Container>
        <GroupTitle>Featured Links</GroupTitle>
        {[0, 0, 0, 0, 0].map((i, index) => (
          <ExternalLink key={index}>
            <Bullet />
            <Label>My Website</Label>
          </ExternalLink>
        ))}

        <GroupTitle>Other Links</GroupTitle>
        {[0, 0, 0, 0, 0].map((i, index) => (
          <ExternalLink key={index}>
            <Bullet />
            <Label>My Website</Label>
          </ExternalLink>
        ))}
      </Container>

      <Divisor />
      <Container>
        <Footer sourceCode={sourceCode} author={author} />
      </Container>
    </ThemeProvider>
  );
}
