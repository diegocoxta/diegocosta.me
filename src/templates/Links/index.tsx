import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Links from './Links';

export default (): React.ReactElement => {
  const {
    site: {
      siteMetadata: {
        metatags: { author, description, avatar },
      },
    },
  } = useStaticQuery(graphql`
    query LinksTemplateQuery {
      site {
        siteMetadata {
          metatags {
            author
            description
            avatar
          }
        }
      }
    }
  `);

  return <Links author={author} description={description} avatar={avatar} />;
};
