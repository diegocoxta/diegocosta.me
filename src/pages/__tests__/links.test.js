import React from 'react';
import { render } from '@testing-library/react';

import { useStaticQuery, mockUseStaticQuery } from '~/__mocks__/graphql';
import Links from '../links';

jest.mock('../../components/Footer', () => () => <p>Footer</p>);

describe('<Links>', () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders properly', () => {
    const { baseElement } = render(<Links />);
    expect(baseElement).toMatchSnapshot();
  });
});
