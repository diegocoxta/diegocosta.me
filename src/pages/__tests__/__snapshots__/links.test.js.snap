// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links> renders properly 1`] = `
.c2 {
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;
}

.c2:focus {
  outline: none;
}

.c3 {
  font-size: 6vw;
  color: #2F3D4F;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

.c4 {
  color: #D3BF9C;
}

.c5 {
  background: linear-gradient(90deg, #ffffff 20px, transparent 1%) center,linear-gradient(#ffffff 20px, transparent 1%) center,#2F3D4F;
  background-size: 22px 22px;
  height: 120px;
  margin: 40px 0;
  width: 100%;
}

.c0 {
  max-width: 660px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  width: 90px;
  height: 90px;
}

.c6 {
  text-transform: uppercase;
  text-align: center;
  font-size: 16px;
  font-weight: 700;
  margin-bottom: 20px;
}

.c7 {
  border: 1px solid #d4d4d8;
  padding: 5px;
  border-radius: 100px;
  display: block;
  display: flex;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.c8 {
  width: 50px;
  height: 50px;
  border-radius: 50px;
  background-color: #D3BF9C;
}

.c9 {
  margin: auto 10px;
}

@media (min-width: 760px) {
  .c3 {
    font-size: 40px;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <img
        class="c1"
      />
      <a
        class="c2"
        href="/"
      >
        <h1
          class="c3"
        >
          Diego
          <span
            class="c4"
            data-testid="header-lastname"
          >
            C
            .
          </span>
        </h1>
      </a>
    </section>
    <div
      class="c5"
    />
    <section
      class="c0"
    >
      <h2
        class="c6"
      >
        Featured Links
      </h2>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <h2
        class="c6"
      >
        Other Links
      </h2>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
      <a
        class="c7"
      >
        <div
          class="c8"
        />
        <p
          class="c9"
        >
          My Website
        </p>
      </a>
    </section>
    <div
      class="c5"
    />
    <section
      class="c0"
    >
      <p>
        Footer
      </p>
    </section>
  </div>
</body>
`;
